events/controller/task.go

```diff
@@ -280,3 +283,96 @@ func (n *NatsConditionTaskRepository) Query(ctx context.Context) (*condition.Tas
 
 	return task, nil
 }
+
+// HTTPTaskRepository implements the ConditionTaskRepository to Publish and Query Task information from NATS over HTTP.
+type HTTPTaskRepository struct {
+	logger        *logrus.Logger
+	orcQueryor    orc.Queryor
+	conditionKind condition.Kind
+	conditionID   uuid.UUID
+	serverID      uuid.UUID
+}
+
+func NewHTTPTaskRepository(
+	serverID,
+	conditionID uuid.UUID,
+	conditionKind condition.Kind,
+	orcQueryor orc.Queryor,
+	logger *logrus.Logger,
+) ConditionTaskRepository {
+	return &HTTPTaskRepository{
+		orcQueryor:    orcQueryor,
+		conditionID:   conditionID,
+		conditionKind: conditionKind,
+		serverID:      serverID,
+		logger:        logger,
+	}
+}
+
+// Publish implements the ConditionTaskRepository interface to record Task information in the Task KV.
+func (h *HTTPTaskRepository) Publish(ctx context.Context, task *condition.Task[any, any], tsUpdateOnly bool) error {
+	_, span := otel.Tracer(pkgName).Start(
+		ctx,
+		"controller.task_http.Publish",
+		trace.WithSpanKind(trace.SpanKindConsumer),
+	)
+	defer span.End()
+
+	resp, err := h.orcQueryor.ConditionTaskPublish(ctx, h.conditionKind, h.serverID, h.conditionID, task, tsUpdateOnly)
+	if err != nil {
+		h.logger.WithError(err).Error("Task update error")
+		return errors.Wrap(errTaskPublish, err.Error())
+	}
+
+	if resp.StatusCode != 200 {
+		// nolint:goerr113 // I'd like to keep the error definition close to where its in use.
```

Comment:
Why though? I dont see any issue with putting this at the top. Also you are creating this multiple times.

Reply:
I guess I see the issue with moving this up. I love Go's errors, huge improvement over other langauges, but they are not perfect. Go errors are still kinda messy =/

Reply:
fair point, added an error type so its not duplicated [b64fbdf](https://github.com/metal-toolbox/rivets/pull/73/commits/b64fbdf214736e3ea9b1c256a362317c56d8c0e6)